pkg/v3/tickers/time.go

```diff
@@ -38,13 +40,13 @@ func NewTimeTicker[T any](interval time.Duration, observer observer[T], getterFn
 // Start uses the provided context for each call to the getter function with the
 // configured interval as a timeout. This function blocks until Close is called
 // or the parent context is cancelled.
-func (t *timeTicker[T]) Start(pctx context.Context) error {
-	ctx, cancel := context.WithCancel(pctx)
-	defer cancel()
-
-	if !t.closer.Store(cancel) {
-		return fmt.Errorf("already running")
+func (t *timeTicker[T]) Start(ctx context.Context) error {
+	if err := t.StartOnce("timeTicker", func() error { return nil }); err != nil {
+		return err
 	}
+	defer close(t.done)
+	ctx, cancel := t.stopCh.Ctx(ctx)
+	defer cancel()
```

Comment:
can't comment on the exact lines, 
in lines 64-66 (in the for loop) should there be a similar statement as above file 
```
if ctx.Err() != nil {
 					return nil
 				}
```

Reply:
Updated to `continue`

Reply:
yeah that does seem like a bug. thanks for fixing!

Reply:
Had to fix a test too